vendor/github.com/openshift/library-go/pkg/operator/staticpod/controller/installer/installer_controller.go

```diff
@@ -47,16 +49,66 @@ type InstallerController struct {
 
 	// installerPodImageFn returns the image name for the installer pod
 	installerPodImageFn func() string
+
+	// installStateFunc derives the static pod install state of a node
+	staticPodStateFunc StaticPodStateFunc
+}
+
+// StaticPodState is the status of a static pod that has been installed to a node.
+type StaticPodState int
+
+const (
+	// staticPodStatePending means that the installed static pod is not up yet.
+	StaticPodStatePending = StaticPodState(iota)
+	// StaticPodStateReady means that the installed static pod is ready.
+	StaticPodStateReady
+	// StaticPodStateFailed means that the static pod installation of a node has failed.
+	StaticPodStateFailed
+)
+
+const deploymentIDLabel = "deployment-id"
+
+// StaticPodStateFunc derives the status of the installed static pod. It returns the state, an array of error strings
+// in case of StaticPodStateFailed and an error in case the state couldn't be checked.
+type StaticPodStateFunc func(targetDeploymentID int32, nodeName string) (StaticPodState, []string, error)
```

Comment:
why we need this nested function stuff?

Reply:
why nested?

Reply:
> why we need this nested function stuff?

I think he means the closure.  Same distaste as me I suspect :)

Reply:
I am a fan of closures. Luckily Golang has them.